accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -901,6 +904,53 @@ These are extended as
 per
 [separate proposal](https://github.com/dart-lang/language/blob/master/resources/type-system/flow-analysis.md).
 
+### The main function
+
+The section 'Scripts' in the language specification is replaced by the
+following:
+
+Let _L_ be a library that exports a declaration _D_ named `main`.  It is a
```

Comment:
To make sure I understand - are we considering going forward with a breaking change in the implementations and applying these restrictions to _all_ declarations named `main`?

Reply:
My intention was to make these changes applicable to code with null safety only, by putting them into the null safety spec.

In the past, we have considered changes that may break opted-in code as non-breaking, but we will of course need to consider them breaking when the amount of pre-existing opted-in code is large enough, and in particular when null safety has been released as a stable feature.

At this point I think it is still fair to say that it is not a breaking change.

Reply:
> At this point I think it is still fair to say that it is not a breaking change.

I think the edge where it would impede a migration to null safety (if there is a package with a non-conforming `main` in the public API) is so unlikely that's probably safe to consider it that way.